src/main/java/in/ashwanthkumar/gocd/github/util/URLUtils.java

```diff
@@ -0,0 +1,18 @@
+package in.ashwanthkumar.gocd.github.util;
+
+import java.net.URL;
+import java.net.URLConnection;
+
+public class URLUtils {
+    public boolean isValidURL(String url) {
+        boolean isValid = true;
+        try {
+            URL urlObj = new URL(url);
+            URLConnection conn = urlObj.openConnection();
+            conn.connect();
```

Comment:
The second answer says if we cast it to `HttpURLConnection` then there's a `#disconnect()`. Let me do some benchmarking on number of CLOSE_WAIT connections when used on a large enough sample. That would answer if the sockets are getting closed or not.